components/search/DateFilter.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <b-form inline>
+    <legend>
+      {{ name }}
+    </legend>
+    <label
+      class="sr-only"
+      for="start-date"
+    >
+      {{ $t('dateFilter.startDate') }}
+    </label>
+    <b-input
+      id="start-date"
+      v-model="form.start"
+      class="mb-2 mr-sm-2 mb-sm-0"
+      type="date"
+    />
+
+    <label
+      class="sr-only"
+      for="end-date"
+    >
+      {{ $t('dateFilter.endDate') }}
+    </label>
+    <b-input
+      id="end-date"
+      v-model="form.end"
+      type="date"
+    />
+  </b-form>
+</template>
+
+<script>
+  export default {
+    name: 'DateFilter',
+
+    props: {
+      name: {
+        type: String,
+        required: true
+      },
+
+      start: {
+        type: String,
+        default: ''
+      },
+
+      end: {
+        type: String,
+        default: ''
+      }
+    },
+
+    data() {
+      return {
+        form: {
+          name: this.name,
+          start: this.start,
+          end: this.end
+        }
+      };
+    },
+
+    watch: {
+      form: {
+        deep: true,
+        handler() {
+          this.$emit('dateFilter', this.form);
```

Comment:
This should emit name as a distinct parameter first, but the final parameter is fine to be an object with start and end properties.

I don't think it needs to go on watch either. It should be possible to just observe the `change` event on each date input and have that call a method which emits the required data.